### ISSUE TITLE
Loss and Spec

### ISSUE THREAD

**Issue by blackbird-fish** [ms-dot-k/Visual-Context-Attentional-GAN#6]
I have trained for 24k steps , but the spec did not get a good restoration, the loss did not  start to converge
![image](https://github.com/ms-dot-k/Visual-Context-Attentional-GAN/assets/68136663/44376727-9d7b-4433-a67c-12c1824ae053)
![image](https://github.com/ms-dot-k/Visual-Context-Attentional-GAN/assets/68136663/dc503dd3-c8d3-4569-acd1-7dbee9ee1a64)

**Comment by ms-dot-k** [comment]
Maybe you are trying to train the model on a different dataset.

Are many of the training data zero padded as shown in the figure? Then, it would be better to calculate the loss (especially reconstruction loss) for the non-padded region only. Or simply upscale the loss weight ("" part shown below) for the reconstruction loss part. You should try different hyperparameters. I recommend checking with 500 first.
`gen_loss = g_loss + recon_loss * "(100~1000.0)" + g_sync_loss`

**Comment by blackbird-fish** [comment]
yes, it is on a different dataset. I think you mentioned the key problem, I will try to upscale the loss weight.

**Comment by blackbird-fish** [comment]
Does this model support multiple speakers? I guess it can just be trained on single speaker

**Comment by ms-dot-k** [comment]
You can find the pre-trained models on LRS2 and LRS3.
So it can be trained on multiple speakers, but the performance is not good as there is no special consideration (module) for it.

**Comment by blackbird-fish** [comment]
Thank for your answer. Maybe it is the reason for the slow converge, and i will try to add the speaker embedding

**Comment by blackbird-fish** [comment]
Hello, I tried to train on GRID and it works .I am confused about why it cannot work on my own dataset.Maybe something wrong with my preprocess, how do you get the 98 points landmark? Is there any reference project

**Comment by ms-dot-k** [comment]
As the model is not originally introduced for sentence-level lip-to-speech synthesis in the wild. It may be hard to achieve proper performance on your own dataset if the dataset is too hard. Note the WER on LRS2 is 109.91%.
The 98 points can be obtained using a landmark detector trained on WFLW(https://github.com/HRNet/HRNet-Facial-Landmark-Detection).

**Comment by blackbird-fish** [comment]
> As the model is not originally introduced for sentence-level lip-to-speech synthesis in the wild. It may be hard to achieve proper performance on your own dataset if the dataset is too hard. Note the WER on LRS2 is 109.91%. The 98 points can be obtained using a landmark detector trained on WFLW(https://github.com/HRNet/HRNet-Facial-Landmark-Detection).

LRS2 is hard to train too, compared with GIRD. The sentences in GIRD are slightly different, thus the model always train the same part. I  guess that's why GRID performs well.
